modem: hl7800: CTS to monitor sleep

rerickson1 opened the pull request:
Use CTS input to monitor HL7800 sleep state
when in SLEEP mode.
When CTS is high, shutdown the UART to
save power.

rerickson1 commented:
This PR depends on https://github.com/zephyrproject-rtos/zephyr/pull/42596. Will rebase once that PR is settled.

ycsin commented on drivers/modem/hl7800.c:
```diff
@@ -4786,20 +4786,67 @@ void mdm_gpio6_callback_isr(const struct device *port, struct gpio_callback *cb,
+	return state1;
+}
+
+void mdm_uart_cts_callback(const struct device *port, struct gpio_callback *cb, uint32_t pins)
```
All of the args here seem unused, maybe add `ARG_UNUSED`?

rerickson1 committed:
modem: hl7800: CTS to monitor sleep

Use CTS input to monitor HL7800 sleep state
when in SLEEP mode.
When CTS is high, shutdown the UART to
save power.

Signed-off-by: Ryan Erickson <ryan.erickson@lairdconnect.com>
